dlhub_sdk/utils/publish.py

```diff
@@ -0,0 +1,244 @@
+
+    Parameters
+    ----------
+    task : dict
+        A dict of the task to publish
+    container_uuid:  str
+    funcx_client:  the funcx client to register with
+
+    Returns
+    -------
+    str
+        The funcX function id
+    """
+
+    fxc = funcx_client
+
+    description = f"A container for the DLHub model {task['dlhub']['shorthand_name']}"
+    try:
+        description = task['datacite']['descriptions'][0]['description']
+    except:
+        # It doesn't have a simple description
+        pass
+    # I believe that the Container Service registers the container w/ funcx
+    # Register the container with funcX
+    # container_id = fxc.register_container(task['dlhub']['ecr_uri'], 'docker', name=task['dlhub']['shorthand_name'],
+    #                                      description=description)
+
+    # Register a function
+    funcx_id = fxc.register_function(dlhub_run, function_name=task['dlhub']['name'],
+                                     container_uuid=container_uuid, description=description, public=True)
+
+    # Whitelist the function on DLHub's endpoint
+    #endpoint_uuid = '86a47061-f3d9-44f0-90dc-56ddc642c000'
+    endpoint_uuid = '2238617a-8756-4030-a8ab-44ffb1446092'
+
+    res = fxc.add_to_whitelist(endpoint_uuid, [funcx_id])
+    print(res)
+    return funcx_id
+
+
+def dlhub_run(event):
+    """Invoke the DLHub servable"""
+    import json
+    import time
+    import os
+
+    from os.path import expanduser
+    path = expanduser("~")
+    os.chdir(path)
+
+    # Check to see if event is from old client
+    if 'data' in event:
+        raise ValueError('Upgrade your DLHub SDK to a newer version: pip install -U dlhub_sdk')
+
+    start = time.time()
+    global shim
+    if "shim" not in globals():
+        from home_run import create_servable
+        with open("dlhub.json") as fp:
+            shim = create_servable(json.load(fp))
+    x = shim.run(event["inputs"],
+                 debug=event.get("debug", False),
+                 parameters=event.get("parameters", None))
+    end = time.time()
+    return (x, (end-start) * 1000)
+
+
+def convert_dict(data, conversion_function=str):
+    if type(data) is dict:
+        string_dict = {}
+        for k, v in data.items():
+            if type(v) is dict:
+                string_dict[k] = convert_dict(v, conversion_function)
+            elif type(v) is list:
+                string_dict[k] = [convert_dict(item, conversion_function) for item in data[k]]
+            else:
+                string_dict[k] = conversion_function(v)
+        return string_dict
+    elif type(data) is list:
+        return [convert_dict(item, conversion_function) for item in data]
+    else:
+        return conversion_function(data)
+
+
+def get_dlhub_file(repository):
+    """
+    Use the github rest api to ensure the dlhub.json file exists.
+
+    :param repository:
+    :return:
+    """
+
+    repo = repository.replace("https://github.com/", "")
```

Comment:
some clarity on how this is parsing -- looks like it's trying to get <username/reponame>?

Reply:
Yes, it is parsing the repository URL to get the usernamne/reponame (or orgname/reponame) as this is what the Github sdk expects for Github.get_repo()

Reply:
great! if you can just add a clarifying comment, I'll resolve this